src/compiler/sys.ts

```diff
@@ -569,7 +573,31 @@ namespace ts {
                     }
                 },
                 setTimeout,
-                clearTimeout
+                clearTimeout,
+                require(initialDir: string, moduleName: string) {
+                    // http://stackoverflow.com/questions/40366008/how-do-i-load-a-node-module-as-if-the-current-module-were-located-somewhere-else
+                    const mod = module as { paths: string[] };
+                    if (mod === undefined) {
+                        return { module: undefined, error: { message: "Module object is not defined" } };
+                    }
+
+                    if (mod.paths === undefined) {
+                        return { module: undefined, error: { message: "Module object has no 'paths' property" } };
+                    }
+
+                    // Note: 'require' needs the path to be all forward slashes and absolute
+                    initialDir = normalizeSlashes(nodeSystem.resolvePath(initialDir));
+                    mod.paths.unshift(initialDir);

```

Comment:
Definitely aware (see earlier comment in this function for a link to an SO question about this). I considered this to be preferable to re-implementing their module resolution algorithm since we don't necessarily know how it works in all cases. Thoughts?